clients/go/entities/job_test.go

```diff
@@ -15,72 +15,81 @@
 package entities
 
 import (
-	"github.com/zeebe-io/zeebe/clients/go/pb"
-	"reflect"
 	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/zeebe-io/zeebe/clients/go/pb"
 )
 
-type MyType struct {
+type testType struct {
 	Foo   string
 	Hello string
 }
 
 var (
-	expectedJson      = "{\"foo\": \"bar\", \"hello\": \"world\"}"
-	expectedJsonAsMap = map[string]string{
-		"foo":   "bar",
-		"hello": "world",
-	}
-	expectedJsonAsMapVariables = map[string]interface{}{
-		"foo":   "bar",
-		"hello": "world",
-	}
-	expectedJsonAsStruct = MyType{
+	job = Job{pb.ActivatedJob{
+		CustomHeaders: `{"foo": "bar", "hello": "world"}`,
+		Variables:     `{"foo": "bar", "hello": "world"}`,
+	}}
+	wantStruct = testType{
 		Foo:   "bar",
 		Hello: "world",
 	}
-	job = Job{pb.ActivatedJob{
-		CustomHeaders: expectedJson,
-		Variables:     expectedJson,
-	}}
 )
 
 func TestJob_GetVariablesAsMap(t *testing.T) {
-	variables, err := job.GetVariablesAsMap()
+	got, err := job.GetVariablesAsMap()
 	if err != nil {
-		t.Error("Failed to get variables as map", err)
+		t.Errorf("job.GetVariablesAsMap() = %v", err)
 	}
-	if !reflect.DeepEqual(variables, expectedJsonAsMapVariables) {
-		t.Error("Failed to get variables as map, got", variables, "instead of", expectedJsonAsMap)
+
+	want := map[string]interface{}{
+		"foo":   "bar",
+		"hello": "world",
+	}
+
+	if diff := cmp.Diff(want, got, cmpopts.SortMaps(func(a, b string) bool { return a < b })); diff != "" {
```

Comment:
Why is `cmpopts.SortMaps` necessary?

Reply:
Normally Go maps don't guarantee any ordering. So for the sake of comparison, required sorting by key.

However, I didn't look into the implementation of `cmp.Equal` to see that the maps are sorted anyway: https://github.com/google/go-cmp/blob/b1c9c4891a6525d98001fea424c8926c6d77bb56/cmp/compare.go#L478.